src/main/java/io/cdap/plugin/servicenow/sink/transform/ServiceNowTransformer.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.servicenow.sink.transform;
+
+import com.google.gson.JsonObject;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.servicenow.util.ServiceNowConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.RoundingMode;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ *  Transforms structured record to JSON object
+ */
+public class ServiceNowTransformer {
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowTransformer.class);
+  private static final String DATE_PATTERN = "yyyy-MM-dd";
+  private static final String DATE_TIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss";
+
+  public JsonObject transform(StructuredRecord record) {
+    List<Schema.Field> fields = Objects.requireNonNull(record.getSchema().getFields(),
+                                                       "Schema fields cannot be empty");
+    JsonObject jsonObject = new JsonObject();
+    for (Schema.Field field : fields) {
+      String fieldName = field.getName();
+      Schema nonNullableSchema = field.getSchema().isNullable() ? field.getSchema().getNonNullable()
+        : field.getSchema();
+      jsonObject.addProperty(field.getName(), String.valueOf(extractValue(fieldName, record.get(field.getName()),
+                                                                          nonNullableSchema)));
+    }
+    return jsonObject;
+  }
+
+  private Object extractValue(String fieldName, Object value, Schema schema) {
+    if (value == null) {
+      // Return 'null' value as it is
+      return null;
+    }
+
+    Schema.LogicalType fieldLogicalType = schema.getLogicalType();
+    // Get values of logical types properly
+    if (fieldLogicalType != null) {
+      switch (fieldLogicalType) {
+        case TIMESTAMP_MILLIS:
+          return extractUTCDateTime(TimeUnit.MILLISECONDS.toMicros((long) value)).toString();
```

Comment:
do we need to convert timestamp and time types to string here? Caller is calling String.valueOf()

Reply:
We're calling `.toString()` at the end. Would it not convert the types to string?

Reply:
Calling .toString() here is not necessary because the caller is converting to String
```java
String.valueOf(extractValue(fieldName, record.get(field.getName())
```

Reply:
Removed `.toString()` 0a1c20a052aca8be095d4e1cde499b71fed53861